react/src/pages/ListItemRouterLink.tsx

```diff
@@ -8,14 +8,17 @@ export interface ListItemRouterLinkProps {
     primary: React.ReactNode;
     to: string;
     children: React.ReactNode;
+    updatePageName: Function;
 }
 
 // https://material-ui.com/guides/composition/#caveat-with-inlining
-export default function ListItemRouterLink({ primary, to, children }: ListItemRouterLinkProps) {
+export default function ListItemRouterLink({ primary, to, children, updatePageName }: ListItemRouterLinkProps) {
     const ForwardedRouterLink = React.useMemo(
         () =>
             React.forwardRef<HTMLAnchorElement>((linkProps, ref) => (
-                <Link ref={ref} to={to} {...linkProps} />
+                <Link ref={ref} to={to} {...linkProps} onClick={(e: React.MouseEvent) => {
+                    updatePageName((e.target as HTMLInputElement).innerText);
+                }}/>
```

Comment:
This actually causes the page state to be set twice if using the sidebar instead of direct links: once when clicking the link above and again when the render function for the route is called. Since the latter occurs in all cases, then the click event isn't necessary.